src/analysis/retail/priest/shadow/modules/talents/DistortedReality.tsx

```diff
@@ -0,0 +1,63 @@
+import TALENTS from 'common/TALENTS/priest';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { CastEvent, DamageEvent } from 'parser/core/Events';
+import Events from 'parser/core/Events';
+import { calculateEffectiveDamage } from 'parser/core/EventCalculateLib';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+import { formatNumber } from 'common/format';
+import InsanityIcon from 'interface/icons/Insanity';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+
+class DistortedReality extends Analyzer {
+  damage = 0;
+  insanitySpent = 0; // extra insanity requred to cast spell
+  multiplierDistortedReality = 0.2; //20%
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS.DISTORTED_REALITY_TALENT);
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(TALENTS.DEVOURING_PLAGUE_TALENT),
+      this.onDevouringPlagueDamage,
+    );
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(TALENTS.DEVOURING_PLAGUE_TALENT),
+      this.onDevouringPlagueCast,
+    );
+  }
+
+  onDevouringPlagueDamage(event: DamageEvent) {
+    this.damage += calculateEffectiveDamage(event, this.multiplierDistortedReality);
+  }
+
+  onDevouringPlagueCast(event: CastEvent) {
+    const resource = event.classResources?.at(0)?.cost; //Some buffs grant free Devouring Plagues, which do not have a resource cost
+    if (resource != null) {
```

Comment:
so this technically works because it is using loose equality (aka `==` instead of `===`). generally, loose equality should be avoided because of how complex its rules are.

this should be either

```suggestion
    if (resource !== undefined) {
```

or

```suggestion
    if (!resource) {
```
(ironic that I recommend this, since it is loose equality still. however, the behavior around `!` is much simpler and harder to mess up)

Reply:
I've updated it to `!== undefined` from `!=null`, and I'll make sure to avoid loose equality in the future.  Also, I also tried `!resource`, but it didn't seem to work the same.